### ISSUE TITLE
Use environment variables from k8s deployment

### ISSUE THREAD

**Issue by larsf96** [microsoft/mindaro#175]
I don't know if this should already work, but I couldn't get it working though.

My scenario is that I try to start a service locally with Bridge2Kubernetes. The service already has a kubernetes deployment and service with some environment variables that are referring to configMaps and secrets.
When I now start the launch config for Bridge2Kubernetes, my service is instantly crashing, because the environment variables could not be resolved.
It would be very nice to just use the existing configuration to not have duplicates of those environment variables locally. 

I've set "useKubernetesServiceEnvironmentVariables" to true, but this had no effect.

Am I doing something wrong or is this feature not yet supported?

**Comment by irperez** [comment]
We've had the same issue.  To compensate, we've setup these variables on the local machine.  Not ideal, but it works.  I wish there was a way that these variables would get setup based on the configmap.  That would make things ideal for sure.

**Comment by pragyamehta** [comment]
@larsf96  @irperez We do support this scenario through a file called KubernetesLocalProcessConfig.yaml
Please go through this link for more details: https://docs.microsoft.com/en-us/visualstudio/containers/configure-bridge-to-kubernetes?view=vs-2019
Environment variables needed by the app can be set here. You can also specify volumes that your pod required which will be downloaded locally when debugging. Let me know if you have any questions.

**Comment by larsf96** [comment]
@pragyamehta Thanks for the hint, but I'm not quite sure if this is exactly what I need.
The article is telling about downloading a volume and setting the path as environment variable. 
What I need is "replicating" the environment variables from the current k8s cluster environment. Most of them are set as configMapValue's in the deployment and I want to use exactly the same env variables without having to setting up those locally.
If that is possible with the above method, I might need a hint on how to do that

**Comment by pragyamehta** [comment]
@larsf96 Apologies for the confusion. It seems that you do not need KubernetesLocalProcessConfig.yaml at all for the scenario you are describing. What you are asking for is supported out of the box for both isolated and non-isolated modes.

Below is the env section of my deployment's container spec:

![image](https://user-images.githubusercontent.com/22248649/119429374-920e7e00-bcc3-11eb-8f3b-a222c424f0b5.png)

I modified the code to print this environment variable in my app code and I can see it getting printed once I debug with Bridge To Kubernetes.
![image](https://user-images.githubusercontent.com/22248649/119429611-047f5e00-bcc4-11eb-995a-1a665fc51cb4.png)

Can you try the scenario again and let me know if it works for you? If not, do provide the repro steps and I will try to unblock you. Thanks!

**Comment by pragyamehta** [comment]
@irperez Hello! Please see my reply above and let us know if it works for you! If not, please provide your repro steps and I will try to unblock you. Thanks!

**Comment by larsf96** [comment]
@pragyamehta Thank you!
I think it was my mistake, looked like I didn't start the service with Bridge to Kubernetes the first time.
Everything is working not :)

**Comment by pragyamehta** [comment]
Closing this as resolved. Please feel free to reach out for any feedback! :)

**Comment by irperez** [comment]
@pragyamehta in our case, the following scenario did not work when we pull environment variables with our configmap.  This pulls out all the variables in the configmap and loads them into the env variables.

![image](https://user-images.githubusercontent.com/1376080/126654508-d15cd270-8c88-44f4-b0a9-fd8d8d858f00.png)

**Comment by pragyamehta** [comment]
@irperez  Hi, I just tried this scenario on VSCode with the latest production bits. Below is my configmap:
![image](https://user-images.githubusercontent.com/22248649/126690741-4cf127af-c150-4da6-a538-b36ee4e3c40d.png)

Below is my deployment showing the envFrom setup:
![image](https://user-images.githubusercontent.com/22248649/126690838-7d588aa0-c65c-4c18-bb27-5dfe6a85ddf9.png)

Below is a screenshot of debugging with Bridge to Kubernetes showing the env var being printed:
![image](https://user-images.githubusercontent.com/22248649/126690934-31bcc787-b888-4986-9984-4390832b9857.png)

Let me know if you have any questions.
